vortex-datafusion/src/persistent/opener.rs

```diff
@@ -28,26 +29,31 @@ impl FileOpener for VortexFileOpener {
         let read_at =
             ObjectStoreReadAt::new(self.object_store.clone(), file_meta.location().clone());
 
-        let mut builder = LayoutReaderBuilder::new(
-            read_at,
-            LayoutDeserializer::new(self.ctx.clone(), Arc::new(LayoutContext::default())),
-        );
+        let deserializer =
+            LayoutDeserializer::new(self.ctx.clone(), Arc::new(LayoutContext::default()));
+
+        let message_cache = Arc::new(RwLock::new(LayoutMessageCache::default()));
+
+        let mut builder = LayoutReaderBuilder::new(read_at.clone(), deserializer.clone())
+            .with_message_cache(message_cache.clone());
 
         if let Some(batch_size) = self.batch_size {
             builder = builder.with_batch_size(batch_size);
         }
 
-        if let Some(predicate) = self
+        let row_filter = self
             .predicate
             .clone()
             .map(convert_expr_to_vortex)
             .transpose()?
-        {
-            builder = builder.with_row_filter(RowFilter::new(predicate));
+            .map(RowFilter::new);
+
+        if let Some(row_filter) = row_filter {
+            builder = builder.with_row_filter(row_filter);
```

Comment:
All this logic should be in the stream or builder

Reply:
Ok, this is because of externalizing the message cache, we should revert that change for now and the diff here disappears